spec/unit/provider/user/useradd_spec.rb

```diff
@@ -44,6 +44,27 @@
       allow(provider).to receive(:exists?).and_return(false)
     end
 
+    it "should not redact the command from debug logs if there is no password" do
+      described_class.has_feature :manages_passwords
```

Comment:
Does this set a class variable that should be cleared between tests?

Reply:
When I ran an experiment in [provider_spec.rb (context "feature with no methods")](https://github.com/puppetlabs/puppet/blob/6.0.x/spec/unit/provider_spec.rb#L839), these tests pass.  I think that means that this setting/variable does not need to be cleared between tests.  What do you think?
```
      context "when one test sets has_feature" do
        it "can set has_feature" do
          subject.has_feature :undemanding
        end
        it { is_expected.not_to be_undemanding }
      end
```

Reply:
Ah in your case there is a unique `subject` for each example. But the test uses `described_class` which refers to the same instance of `Puppet::Provider::User::UseraddProvider`. I manually verified that features are leaking across tests:

```
> described_class.features
[:allows_duplicates, :manages_expiry, :manages_homedir, :manages_shell, :system_users]
```

That said the `has_feature` pattern is used in a bunch of places, so it's not blocking this PR.